src/olink/clientregistry.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "core/olink_common.h"
+#include "core/basenode.h"
+#include <map>
+#include <vector>
+
+
+namespace ApiGear {
+namespace ObjectLink {
+
+class IObjectSink;
+class ClientNode;
+
+/**
+ * A Registry is a global storage to keep track of objects stored as objectSink for the messages 
+ * and a client node associated with that objectSink.
+ * Each object can use only one client node.
+ * Each object is registerd with its id, available with olinkObjectName() call.
+ * This id has to be unique in the registry, only first object with same id will be registerd.
+ * A client node may be used for many objects.
+ * Register your object and a client node separately: an object with addObject function and 
+ * the client node with setNode, which requires also the target objectId.
+ * The order of registration is not relevant.
+ */
+class OLINK_EXPORT ClientRegistry: public LoggerBase {
+public:
+    /**dtor*/
+    virtual ~ClientRegistry() = default;
+
+    /**
+    * Adds ClientNode to registry for objectId.

```

Comment:
You should not explain a function called "set" with the term "add". This either indicated docs are not right or function name is wrong.